File: Animation-OC/Animation-OC/ScaleViewController.m

//
//  ScaleViewController.m
//  Animation-OC
//
//  Created by chenyufeng on 15/12/31.
//  Copyright © 2015年 chenyufengweb. All rights reserved.
//

#import "ScaleViewController.h"

@interface ScaleViewController ()

@property (weak, nonatomic) IBOutlet UIView *redSquare;

@end

@implementation ScaleViewController

- (void)viewDidLoad {

  [super viewDidLoad];
}

- (void)viewWillAppear:(BOOL)animated{

  [super viewWillAppear:animated];
  [UIView animateWithDuration:2 animations:^{
    //宽高缩放比；
    self.redSquare.transform = CGAffineTransformMakeScale(2, 3);
  }];
}

@end

File: PositionAnimation01/PositionAnimation01/ScaleViewController.swift
//
//  ScaleViewController.swift
//  PositionAnimation01
//
//  Created by chenyufeng on 15/8/20.
//  Copyright (c) 2015年 chenyufengweb. All rights reserved.
//

import UIKit

class ScaleViewController: UIViewController {

    
    
    @IBOutlet weak var greenSquare: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(animated)
        
        func anim(){
        
            self.greenSquare.transform = CGAffineTransformMakeScale(0.5, 0.5)//缩小为原来的0.5倍；
        }
        
        
        UIView.animateWithDuration(1, animations: anim)
        
        
        
    }

}












